include/pipettes/core/uart_message_handler.hpp

```diff
@@ -0,0 +1,61 @@
+#pragma once
+
+#include <variant>
+
+#include "common/hal/io.hpp"
+#include "common/hal/motor.hpp"
+#include "pipettes/core/communication.hpp"
+#include "pipettes/core/pipette_messages.hpp"
+
+template <typename T, motor_protocol::MotorProtocol Motor>
```

Comment:
Let's give the first template protocol a better name than T. I think we can also qualify it in the template as we do with motor protocol. Like maybe `template<io::WriterProtocol Writer, motor_procotol::MotorProtocol Motor>`? `T` is the common case for examples because it's so generic and thus useful as a parameter for something like a container, but here I think it's just confusing.